main.cc

```diff
@@ -1379,6 +1382,19 @@ int main(int argc, char*argv[])
       return rtn;
 }
 
+int main(int argc, char*argv[])
+{
+	parse_cmdargs(argc, argv);
+
+	preprocess();
+
+	find_root_module();
```

Comment:
These functions should return error codes, and the main function should check those error codes.